kover-features-java/src/main/java/kotlinx/kover/features/java/KoverLegacyFeatures.java

```diff
@@ -0,0 +1,134 @@
+package kotlinx.kover.features.java;
+
+import com.intellij.rt.coverage.instrument.api.OfflineInstrumentationApi;
+import com.intellij.rt.coverage.report.api.Filters;
+import com.intellij.rt.coverage.report.api.ReportApi;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * Kover Features for support Kover capabilities in Kover CLI via outdated API.
+ */
+public class KoverLegacyFeatures {
+
+    /**
+     * Generate modified class-files to measure the coverage.
+     *
+     * @param resultDir Directory where the instrumented class-files will be placed
+     * @param originalDirs Root directories where the original files are located, the coverage of which needs to be measured
+     * @param filters Filters to limit the classes that will be displayed in the report
+     * @param countHits Flag indicating whether to count the number of executions to each block of code. {@code false} if it is enough to register only the fact of at least one execution
+     */
+    public static void instrument(File resultDir,
+                           List<File> originalDirs,
+                           ClassFilters filters,
+                           boolean countHits
+    ) {
+        ArrayList<File> outputs = new ArrayList<>(originalDirs.size());
+        for (int i = 0; i < originalDirs.size(); i++) {
+            outputs.add(resultDir);
+        }
+
+        String condySetting = ConDySettings.disableConDy();
+        try {
+            OfflineInstrumentationApi.instrument(originalDirs, outputs, convertFilters(filters), countHits);
+        } finally {
+            ConDySettings.restoreConDy(condySetting);
+        }
+    }
+
+    /**
+     * Generate Kover XML report, compatible with JaCoCo XML.
+     *
+     * @param xmlFile path to the generated XML report
+     * @param binaryReports list of coverage binary reports in IC format
+     * @param classfileDirs list of root directories for compiled class-files
+     * @param sourceDirs list of root directories for Java and Kotlin source files
+     * @param title Title for header
+     * @param filters Filters to limit the classes that will be displayed in the report
+     * @throws IOException In case of a report generation error
+     */
+    public static void xmlReport(
```

Comment:
Maybe name methods using verbs? E.g. `writeXmlReport` or `writeXmlReportTo`, so it is clear that `xmlFile` is a desired directory, not some existing report.